### PR TITLE
Add Users with redux

### PR DESCRIPTION
**Pull request by StrangeNoob**
#1

### CODE REVIEW

File: client/src/actionTypes.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const ActionTypes = {
+    AddUser: "ADD_USER",
+};
```
Comment (ankitjena) [inline client/src/actionTypes.js]: Keep these within the action files

File: client/src/components/LoginCallback.js [hunk 2]
```diff
@@ -58,8 +62,10 @@ export function LoginCallback() {
         method: 'post',
         data: userData,
       })
-
-      // eslint-disable-next-line no-undef
+      dispatch({
+        type:ActionTypes.AddUser,
+        userData
+      })
```
Comment (ankitjena) [inline client/src/components/LoginCallback.js]: Make a separate action file, keep the component within it's folder like Login and LoginCallback can be both kept inside Login folder which will have it's set of actions

Comment (ankitjena) [reply client/src/components/LoginCallback.js]: Check [https://github.com/ankitjena/evential/tree/master/client/src](https://github.com/ankitjena/evential/tree/master/client/src) for example

File: client/src/reducer.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    state.users[nextId] = {
+      id: nextId,
+      email: email,
+      name: name,
+      username: username
+    };
+    console.log(state.users[nextId]);
+  },
+});
+
```
Comment (ankitjena) [inline client/src/reducer.js]: Each component will have it's own reducer, and the reducer file at root will merge those

File: client/yarn.lock [hunk 4]
```diff
@@ -905,7 +905,7 @@
   dependencies:
     regenerator-runtime "^0.13.4"
 
-"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.0", "@babel/runtime@^7.5.1", "@babel/runtime@^7.7.4":
+"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.0", "@babel/runtime@^7.5.1", "@babel/runtime@^7.5.5", "@babel/runtime@^7.7.4":
```
Comment (ankitjena) [inline client/yarn.lock]: remove the yarn.lock file and add it to .gitignore

**Comment by ganeshpatro321** [comment]
A small suggestion if you are setting up redux @StrangeNoob :

- Create a container file for each and every component which will have functionalities with the redux state.
- Create a folder named `redux` in the `src/` and have the following things there
- - store configuration
- - A folder `actions` which will contain all the action types.
- - A folder `reducers` which will contain all the  reducers.
- - A folder `selectors` if required.

For any component `views.js` there should be file `viewConatiner.js` which will handle the mapping state and dispatch to props, (that is this file will contain `mapStatetoProps` and `mapDispatchtoProps` functions). Ideally you should create a folder for a component, like creating a folder for `view` component which will contain three file normally:
 - `view.js` : main component.
 - `viewContainer.js` : as explained above.
 - `view.css` : styling 

Coming to the redux thing. Actions and reducers folders should each contain files named `view.js` which will actions and reducers which are related to the view component.

The file strucure should be like this:

```
+--public
+-- src
|     +-- components
|      |    +-- views
|      |     |     +--views.js
|      |     |     +--viewsContainer.js
|      |     |     +--views.css
|     +-- redux
|      |    +-- reducers
|      |     |    +-- views.js 
|      |    +-- actions
|      |     |    +-- views.js 
```


Bahut time lagila lekhibalu :laughing:

**Comment by ankitjena** [comment]
@StrangeNoob Checkout the changes I made

**Pull request merged**